projects/laji/src/app/shared/guards/check-login.guard.ts

```diff
@@ -23,38 +22,24 @@ export class CheckLoginGuard  {
     private platformService: PlatformService
   ) {}
 
-  canActivate(route: ActivatedRouteSnapshot): UrlTree|boolean|Observable<boolean> {
+  canActivate(route: ActivatedRouteSnapshot): UrlTree | boolean | Observable<boolean> {
+    // queryparams removed in SSR
     if (this.platformService.isServer) {
       this.location.replaceState(this.location.path().split('?')[0], '');
       return true;
     }
-    if (!this.isChecked && route.queryParams['token']) {
-      this.isChecked = true;
-      this.location.replaceState(this.location.path().split('?')[0], '');
-      if (this.platformService.canUseWebWorkerLogin) {
-        this.personApi.personFindByToken(PERSON_TOKEN).pipe(
-          map(() => PERSON_TOKEN),
-          catchError(e => {
-            if (e.status === 0) {
-              this.platformService.canUseWebWorkerLogin = false;
-
-              return of(route.queryParams['token']);
-            }
-            return of(PERSON_TOKEN);
-          }),
-          switchMap(token => this.userService.login(token))
-        ).subscribe();
-      } else {
-        this.userService.login(route.queryParams['token']).subscribe();
-      }
-
-      return this.router.parseUrl(this.userService.getReturnUrl());
-    }
 
-    if (this.isChecked) {
+    // continue normally if login has already been checked
+    if (this.isLoginChecked) {
       return true;
     }
 
-    return this.userService.checkLogin();
+    this.isLoginChecked = true;
+
+    // store in localstorage
+    return this.userService.login(route.queryParams['token']).pipe(map(_ => true));
+
+    // TODO go to return url
+    return this.router.parseUrl(this.userService.getReturnUrl());
```

Comment:
The TODO here must be implemented, it's an important feature.

Reply:
Not sure how I missed this... should work now